### PR TITLE
8 post comment

### PR DESCRIPTION
**Pull request by ELR143**

### CODE REVIEW

File: src/components/PostComment.jsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      return [optimisticComment, ...currentComments];
+    });
+
+    postComment(input, article_id)
+      .then((comment) => {
+        setComments((currentComments) => {
+          const resetComments = currentComments.filter((comment) => {
+            return comment.comment_id !== null;
+          });
+          console.log(currentComments);
```
Comment (popatre) [inline src/components/PostComment.jsx:39]: remove log

File: src/components/PostComment.jsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    };
+
+    setComments((currentComments) => {
+      return [optimisticComment, ...currentComments];
+    });
+
+    postComment(input, article_id)
+      .then((comment) => {
+        setComments((currentComments) => {
+          const resetComments = currentComments.filter((comment) => {
```
Comment (popatre) [inline src/components/PostComment.jsx:36]: not sure what this filter does

**Review by popatre — CHANGES_REQUESTED** [review]
- Lovely message to let someone know the post has failed and update the UI accordingly
- nice posting functionality to the api
- Lovely controlled components for the forms

- [ ] We need at least some basic validation on the inputs, at least to stop blank posts
- [ ] Need to disable the form when posting to stop duplicates or some loading pattern to stop someone spamming the post button on a slow connection - can disable the button or have you loading message, replace the button

**Review by nc-poppy-m — APPROVED** [review]
Good work here! Only thing I would say is perhaps don't have the comment box disappear completely on an error, because then the user has to refresh the page to type a comment once their internet is back. However, this is just a bonus UX consideration. Merge when ready

**Pull request merged**